torch/onnx/_internal/fx/dynamo_graph_extractor.py

```diff
@@ -0,0 +1,192 @@
+from __future__ import annotations
+
+import copy
+from typing import Any, Callable, Mapping, Optional, Sequence, Tuple, Union
+
+import torch.onnx
+from torch.onnx import utils as onnx_utils
+from torch.onnx._internal import exporter
+from torch.onnx._internal.fx import io_adapter as io_adapter
+
+
+class DynamoOptimizeGraphCaptureCompiler:
+    def __init__(self):
+        self.captured_graph: Optional["torch.fx.GraphModule"] = None
+        self.captured_graph_count = 0
+
+    def __call__(self, graph_module: "torch.fx.GraphModule", _):
+        assert self.captured_graph_count == 0
+        self.captured_graph = graph_module
+        self.captured_graph_count += 1
+        return graph_module
+
+
+class DynamoOptimize(exporter.FXGraphExtractor):
```

Comment:
Could you put the reason in PR description why we need this back after https://github.com/pytorch/pytorch/pull/99202